src/Compilers/CSharp/Portable/Symbols/AssemblySymbol.cs

```diff
@@ -858,11 +866,31 @@ private NamedTypeSymbol ApplyGenericArguments(NamedTypeSymbol symbol, Type[] typ
                             continue;
                         }
                     }
+                    else if (isWellKnownType && warnings is null)
```

Comment:
>isWellKnownType && warnings is null [](start = 29, length = 35)

Is this how we detect "well-known types after C# 7"? Consider caching this value in a dedicated local before the loop, for clarity.

Reply:
It is not clear to me why ```isWellKnownType``` is required, it feels like the adjusted rules should apply for a case of a not well-known type. 

---
In reply to: [585695313](https://github.com/dotnet/roslyn/pull/51571#discussion_r585695313) [](ancestors = 585695313)